shopinvader_api_address/__manifest__.py

```diff
@@ -13,15 +13,19 @@
         "extendable",
         "extendable_fastapi",
         "fastapi",
-        "shopinvader_address",
+        "sale",
```

Comment:
Why removing shopinvader_address?

Reply:
After starting to move code from "res.partner" to the "shopinvader_api_cart.address_router.helper".

I realize that everything was really related to the router, so at the end I had no more code on "res.partner". I only had an hesitation on the method "_ensure_address_not_used" that may stay on "res.partner"

So at the end the module shopinvader_address had no code, so I remove it.

But maybe I miss some point and some code was think to be reuse outside of the router.

Reply:
> I realize that everything was really related to the router, so at the end I had no more code on "res.partner". I only had an hesitation on the method "_ensure_address_not_used" that may stay on "res.partner"

I respectfully express my dissenting opinion on this matter. From my perspective, the code previously residing in res.partner does not appear to be exclusively tied to the router. Consider a scenario where one opts for a different framework other than FastAPI or has an Odoo module utilizing 'shopinvader addresses' independently of 'shopinvader_api_address.' In such cases, it seems essential to retain all functions originally present in shopinvader_address for broader compatibility and flexibility. This approach ensures that the functionality remains accessible even in diverse usage contexts beyond the current implementation.

Reply:
@sebastienbeau actually `shopinvader_address` is the logic and high level methods that defines what invoicing and delivery addresses mean for shopinvader. There is business logic in there that defines what the invoicing address is what deletion means (= archival) etc.

We can easily imagine that some backend code will want to manipulate such addresses, and that code would then depend on `shopinvader_address` to use these methods, and not `shopinvader_api_address`.

This is the principle that we want to avoid business logic in the API layer as much as possible, and in this case we concluded that the address logic was non-trivial enough to go in its own module.